Packs/Base/Scripts/DBotMLFetchData/DBotMLFetchData.py

```diff
@@ -1,6 +1,1031 @@
+    http_regex = r'http://(?:[a-zA-Z]|[0-9]|[$-_@.&+]|[!*\(\),]|(?:%[0-9a-fA-F][0-9a-fA-F]))+'
+    https_regex = r'https://(?:[a-zA-Z]|[0-9]|[$-_@.&+]|[!*\(\),]|(?:%[0-9a-fA-F][0-9a-fA-F]))+'
+    http_urls = re.findall(http_regex, email_body)
+    https_urls = re.findall(https_regex, email_body)
+    embedded_urls = [a['href'] for a in soup.findAll('a')]
+    all_urls = http_urls + https_urls + embedded_urls
+    all_urls_lengths = [len(u) for u in all_urls]
+    average_url_length = 0 if len(all_urls) == 0 else sum(all_urls_lengths) / len(all_urls)
+    min_url_length = 0 if len(all_urls_lengths) == 0 else min(all_urls_lengths)
+    max_url_length = 0 if len(all_urls_lengths) == 0 else max(all_urls_lengths)
+    shortened_urls_count = len([u for u in all_urls if any(shortened_u in u for shortened_u in SHORTENED_DOMAINS)])
+    drive_count = len([u for u in all_urls if any(drive in u for drive in DRIVE_URL_KEYWORDS)])
+    return {
+        'http_urls_count': len(http_urls),
+        'https_urls_count': len(https_urls),
+        'embedded_urls_count': len(embedded_urls),
+        'all_urls_count': len(all_urls),
+        'average_url_length': average_url_length,
+        'min_url_length': min_url_length,
+        'max_url_length': max_url_length,
+        'shortened_urls_count': shortened_urls_count,
+        'drive_count': drive_count
+    }
+
+
+def get_html_features(soup):
+    global HTML_TAGS
+    html_counter = Counter([tag.name for tag in soup.find_all()])
+    for t in HTML_TAGS:
+        html_counter[t] = 0 if t not in html_counter else html_counter[t]
+    html_counter = {k: v for k, v in html_counter.items() if k in HTML_TAGS}
+    return html_counter
+
+
+def get_ml_features(tokenized_text):
+    global EMBEDDING_DICT
+    if EMBEDDING_DICT is None:
+        with open('/var/glove_top_20k.p', 'rb') as file:
+            EMBEDDING_DICT = pickle.load(file)
+    vectors = [EMBEDDING_DICT[w] for w in tokenized_text if w in EMBEDDING_DICT]
+    if len(vectors) == 0:
+        mean_vector = np.zeros(50)
+    else:
+        mean_vector = np.mean(vectors, axis=0)
+    res = {'glove_avg_{}'.format(i): mean_vector[i].item() for i in range(len(mean_vector))}
+    return res
+
+
+def get_header_value(email_headers, header_name, index=0, ignore_case=False):
+    if ignore_case:
+        header_name = header_name.lower()
+        headers_with_name = [header_dict for header_dict in email_headers if header_dict['name'].lower() == header_name]
+    else:
+        headers_with_name = [header_dict for header_dict in email_headers if header_dict['name'] == header_name]
+    if len(headers_with_name) == 0:
+        return None
+    else:
+        return headers_with_name[index]['value']
+
+
+def get_headers_features(email_headers):
+    spf_result = get_header_value(email_headers, header_name='Received-SPF')
+    if spf_result is not None:
+        spf_result = spf_result.split()[0].lower()
+    else:
+        spf_result = 'other'
```

Comment:
I will say "empty" not other - if the header doesn't exists
also please use const not magic string